simulator/mips/t/unit_test.cpp

```diff
@@ -990,6 +1066,45 @@ TEST_CASE( "MIPS32_instr: jr to 2nd byte (round up to 4th)")
 }
 ////////////////////////////////////////////////////////////////////////////////
 
+TEST_CASE( "MIPS32_instr: lui 0 and 0x4d2")
+{
+    CHECK(MIPS32Instr(0x3c1104d2).get_disasm() == "lui $s1, 0x4d2");
+
+    MIPS32Instr instr( "lui");
+    instr.set_v_src( 0, 0);
+    instr.set_v_imm( 0x4d2);
+    instr.execute();
+    CHECK( instr.get_v_dst()  == 0x04d20000);
+}
+
+TEST_CASE( "MIPS32_instr: lui 0 and 1")
+{
+    MIPS32Instr instr( "lui");
+    instr.set_v_src( 0, 0);
+    instr.set_v_imm( 1);
+    instr.execute();
+    CHECK( instr.get_v_dst()  == 0x10000);
+}
+
+TEST_CASE( "MIPS32_instr: lui 0xffffffff and 0")
+{
+    MIPS32Instr instr( "lui");
+    instr.set_v_src( 0xffffffff, 0);
```

Comment:
What is the point of initializing the source register? LUI does not take care about it.

Reply:
Misunderstanding. I'll rework lui.